builder.go

```diff
@@ -78,6 +81,11 @@ func (b *Builder) Build() (*BuildMetadata, error) {
 	var bom []BOMEntry
 	var slices []Slice
 
+	currentUser := os.Getuid()
```

Comment:
Is this a separate feature or somehow necessary for this to work?

Reply:
`pack` is going to run `creator` as `root`, when exporting to the daemon. Therefore we need to fix permissions and explicitly run buildpacks as an unprivileged user.